src/Mvc/Mvc.Core/src/DependencyInjection/ApiBehaviorOptionsSetup.cs

```diff
@@ -10,34 +10,38 @@
 
 namespace Microsoft.Extensions.DependencyInjection
 {
-    internal class ApiBehaviorOptionsSetup :
-        IConfigureOptions<ApiBehaviorOptions>,
-        IPostConfigureOptions<ApiBehaviorOptions>
+    internal class ApiBehaviorOptionsSetup : IConfigureOptions<ApiBehaviorOptions>
     {
-        internal static readonly Func<ActionContext, IActionResult> DefaultFactory = DefaultInvalidModelStateResponse;
-        internal static readonly Func<ActionContext, IActionResult> ProblemDetailsFactory =
-            ProblemDetailsInvalidModelStateResponse;
-
         public void Configure(ApiBehaviorOptions options)
         {
             if (options == null)
             {
                 throw new ArgumentNullException(nameof(options));
             }
 
-            options.InvalidModelStateResponseFactory = DefaultFactory;
+            options.InvalidModelStateResponseFactory = ProblemDetailsInvalidModelStateResponse;
             ConfigureClientErrorMapping(options);
-        }
 
-        public void PostConfigure(string name, ApiBehaviorOptions options)
-        {
-            // We want to use problem details factory only if
-            // (a) it has not been opted out of (SuppressMapClientErrors = true)
-            // (b) a different factory was configured
-            if (!options.SuppressMapClientErrors &&
-                object.ReferenceEquals(options.InvalidModelStateResponseFactory, DefaultFactory))
+            IActionResult ProblemDetailsInvalidModelStateResponse(ActionContext context)
             {
-                options.InvalidModelStateResponseFactory = ProblemDetailsFactory;
+                var problemDetails = new ValidationProblemDetails(context.ModelState)
+                {
+                    Status = StatusCodes.Status400BadRequest,
+                };
+
+                if (options.ClientErrorMapping.TryGetValue(400, out var clientErrorData))
```

Comment:
This is the new bit. ValidationProblemDetails returned by the ModelStateInvalidFilter should now have a `"type": "https://tools.ietf.org/html/rfc7231#section-6.5.1"` (or whatever Link the user configured).

Reply:
I realized I'm missing a test to verify we use  the user configured value. I'll add one

Reply:
oke